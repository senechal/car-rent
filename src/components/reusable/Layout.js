import Base from './Base';

export const Layout = (props) => <Base as="section" defaultCls="rac-layout" {...props} />;
export const CenterLayout = (props) => <Layout className="rac-layout-center" {...props} />;
export const Header = (props) => <Base as="header" defaultCls="rac-layout-header" {...props}/>;
export const Footer = (props) => <Base as="footer" defaultCls="rac-layout-footer" {...props} />;
export const Sider = (props) => <Base as="aside" defaultCls="rac-layout-sider" {...props} />;
export const Content = (props) => <Base as="main" defaultCls="rac-layout-content" {...props} />;

Layout.CenterLayout = CenterLayout;
Layout.Header = Header;
Layout.Footer = Footer;
Layout.Sider = Sider;
Layout.Content = Content;

export default Layout;
