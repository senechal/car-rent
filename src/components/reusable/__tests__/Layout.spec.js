import { render } from '@testing-library/react';
import * as components from '../Layout';

describe('Testing Layout Components', () => {

  Object.keys(components).forEach((componentName) => {
    it(`should render ${componentName} correctly`, () => {
        const Component = components[componentName];
        const { container } = render(<Component />);
        expect(container).toMatchSnapshot();
    });
  });
});