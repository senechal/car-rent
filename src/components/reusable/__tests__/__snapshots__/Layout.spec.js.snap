// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Layout Components should render CenterLayout correctly 1`] = `
<div>
  <section
    class="rac-layout rac-layout-center"
  />
</div>
`;

exports[`Testing Layout Components should render Content correctly 1`] = `
<div>
  <main
    class="rac-layout-content"
  />
</div>
`;

exports[`Testing Layout Components should render Footer correctly 1`] = `
<div>
  <footer
    class="rac-layout-footer"
  />
</div>
`;

exports[`Testing Layout Components should render Header correctly 1`] = `
<div>
  <header
    class="rac-layout-header"
  />
</div>
`;

exports[`Testing Layout Components should render Layout correctly 1`] = `
<div>
  <section
    class="rac-layout"
  />
</div>
`;

exports[`Testing Layout Components should render Sider correctly 1`] = `
<div>
  <aside
    class="rac-layout-sider"
  />
</div>
`;

exports[`Testing Layout Components should render default correctly 1`] = `
<div>
  <section
    class="rac-layout"
  />
</div>
`;
